server/socket/socketHelperFunctions.js

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable no-unused-vars */
+require('express');
+const mongoose = require('mongoose');
+const ssotToRobotParser = require('../services/SsotToRobotParsing/SsotToRobotParser.js');
+const ssotModels = require('../models/singleSourceOfTruthModel.js');
+const userAccessObject = require('../models/userAccessObjectModel.js');
+const jobsModel = require('../models/robotJobModel.js');
+
+mongoose.set('useFindAndModify', false);
+
+/**
+ * @description Fetches the ssot from the database and parses the ssot to robot code
+ * @param {String} robotId the id of the robot we want the robot code for
+ */
+exports.getRobotCode = async (robotId) => {
+  try {
+    mongoose.connect(process.env.MONGODB_URI, {
```

Comment:
mongoose connect should only be called once not necessarily for each method. This was one of the causes why we switched

Reply:
yes, if mongoose is imported and a connection has already been created then there is no need for a new connection